oauth2_http/java/com/google/auth/oauth2/AwsCredentials.java

```diff
@@ -103,13 +107,15 @@ public class AwsCredentials extends ExternalAccountCredentials {
       this.awsSecurityCredentialsProvider =
           new ProgrammaticAwsSecurityCredentialsProvider(
               builder.awsSecurityCredentialsSupplier, this.awsRegion);
+      this.builtWithSupplier = true;
```

Comment:
Hmm, we shouldn't have to track this. Why not handle this in the builder so we can set the provider directly and not have to handle this logic here?

Reply:
Figured out a better way to handle in the builder, we can just check whether the builder has a credential source, and only set the supplier if it doesn't